orttraining/orttraining/test/graph/optimizer_graph_builder_test.cc

```diff
@@ -199,7 +199,7 @@ static void TestOptimizerGraphBuilderWithInitialStates(OptimizerGraphConfig conf
   const ONNX_NAMESPACE::TensorProto* tensor{};
   for (auto& weight_item : opt_initializer_names_map) {
     for (auto& opt_item : weight_item.second) {
-      ASSERT_TRUE(graph.GetInitializedTensor(opt_item.second, tensor));
+      ORT_ENFORCE(graph.GetInitializedTensor(opt_item.second, tensor));
```

Comment:
>ORT_ENFORCE [](http://example.com/codeflow?start=6&length=11)

It is the same thing, is it not?

Reply:
I think the main problem is the compiler doesn’t know tensor can’t be null when this function returns true.

Reply:
ASSERT_TRUE won't. this is not the assert macro in stdlib. ASSERT_TRUE is from gtest. It won't go away because we still need to run such tests in release mode.